lib/preprocessing_function_similarity.py

```diff
@@ -0,0 +1,24 @@
+from pyspark.sql.functions import monotonically_increasing_id, row_number
+from pyspark.sql.window import Window
+from typing import List
+from pyspark.sql import DataFrame
+from pyspark.ml.feature import VectorAssembler
+
+def DenseVectorToSimilarity(
+    df: DataFrame,
+    columns: List[str],
+    ) -> DataFrame:
+    "Funkcja, która zamienia wybrane kolumny na gotowe do wrzucenia do Similarity, a resztę pozostawia bez zmian"
+
+    df_similarity = df.select(columns)
+    df_rest = df.drop(*columns)
+    vecAssembler2 =  VectorAssembler(inputCols=df_similarity.columns, outputCol="features", handleInvalid="keep")
+    vecAssembler2 = vecAssembler2.transform(df_similarity)
```

Comment:
to możesz skrócić i pozbyć się dodatkowych zmiennych pisząc VectorAssembler(inputCols=df_similarity.columns, outputCol="features", handleInvalid="keep").transform(df_similarity).select(["features"])

Reply:
zrobione